presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingConfigurationProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache.alluxio;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.hive.DynamicConfigurationProvider;
+import com.facebook.presto.hive.HdfsContext;
+import org.apache.hadoop.conf.Configuration;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+
+import static com.facebook.presto.cache.CacheType.ALLUXIO;
+
+public class AlluxioCachingConfigurationProvider
+        implements DynamicConfigurationProvider
+{
+    private final CacheConfig cacheConfig;
+    private final AlluxioCacheConfig alluxioCacheConfig;
+
+    @Inject
+    public AlluxioCachingConfigurationProvider(CacheConfig cacheConfig, AlluxioCacheConfig alluxioCacheConfig)
+    {
+        this.cacheConfig = cacheConfig;
+        this.alluxioCacheConfig = alluxioCacheConfig;
+    }
+
+    @Override
+    public void updateConfiguration(Configuration configuration, HdfsContext context, URI uri)
+    {
+        if (cacheConfig.isCachingEnabled() && cacheConfig.getCacheType() == ALLUXIO) {
```

Comment:
`cacheConfig.isCachingEnabled()` should be a validation than in `if`

Reply:
fixed